Tests/TestCase.php

```diff
@@ -2,6 +2,18 @@
 
 namespace Noxlogic\RateLimitBundle\Tests;
 
-class TestCase extends \PHPUnit_Framework_TestCase
-{
+if (!class_exists('\\PHPUnit\\Framework\\TestCase')) {
```

Comment:
Can't we just drop old phpunit? Anyway is an dev dependency

Reply:
same problem as above I guess, the need for php 5.4 kinda locks you into the old phpunit...

Reply:
I think we need to create a new major version if we want to drop 5.x support?

Reply:
I mean I could move it to PHP 7.1 (since phpunit 7 will reach End of life next week)

Reply:
A major is not required here since php unit is a dev dependency, users are not impacted by it at all.

Consider also this post by the doctrine team that explains why dropping old versions is not considered a BC brak 
https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html

Reply:
(and I agree with them)

Reply:
Hmmm yeah interesting take on it, if you look at it like that it makes sense

Reply:
Ok, the old versions of the Framework bundle still use the Legacy PHPUnit stuff, so anything below framework bundle 4.2 will always fail tests with a new PHPUnit version.
So we'd need to crank up minimum Symfony version to 4.2 as well

Reply:
Not sure if that counts as major version upgrade though

Reply:
Tbh at that point I'd go as far as limiting it to >= 4.3 since 4.2 can't seem to decide which event/dispatcher systems it wants.

Reply:
4.2 is not supported anymore and upgrading to 4.4 should be trivial so for me is a big +1

Reply:
If was up to me will do 3.4, 4.4 and 5.0

Reply:
I made it 3.4 and >= 4.3 only and now the only things that fail are the tests flagged `prefer-lowest ` and the SF5 one (cause the oauth server bundle isn't there yet) so allowed those to fail for now